app/test_main.py

```diff
@@ -1 +1,68 @@
-# write your code here
+from unittest import mock
+import pytest
+
+
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "valid_url, valid_internet, expected_result",
+    [
+        (True, True, "Accessible"),
+        (True, False, "Not accessible"),
+        (False, True, "Not accessible"),
+        (False, False, "Not accessible"),
+    ],
+    ids=[
+        "access only with valid url and valid internet",
+        "no access without internet",
+        "no access without valid url",
+        "no access without valid url and internet"
+    ]
+)
+@pytest.fixture()
+def mocked_url() -> None:
+    with mock.patch("app.main.valid_google_url") as mocked_url:
+        yield mocked_url
+
+
+@pytest.fixture()
+def mocked_internet() -> None:
+    with mock.patch("app.main.has_internet_connection") as mocked_time:
+        yield mocked_time
+
+
+def test_can_access_google_page_with_internet_and_valid_url(
+        mocked_url: mock.MagicMock,
+        mocked_internet: mock.MagicMock
+) -> None:
+    mocked_url.return_value = True
```

Comment:
You added parameters but you still hardcode return values here.
The correct implementation will look something like this:
1. Create one test function with assert that will check for correct result and asserts that mock function has been called.
2. Add parametrizer to that test and use values from parametrizer instead of hard coding.
3. You should mock required functions btw.